sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -166,6 +170,81 @@ class RocksDB(
   @GuardedBy("acquireLock")
   @volatile private var acquiredThreadInfo: AcquiredThreadInfo = _
 
+  // This is accessed and updated only between load and acquire
+  // which means it is implicitly guarded by acquireLock
+  private val colFamilyNameToIdMap = new ConcurrentHashMap[String, Short]()
+
+  private val defaultColumnFamilyIdMapping =
+    Map(StateStore.DEFAULT_COL_FAMILY_NAME -> StateStore.DEFAULT_COL_FAMILY_ID).asJava
+
+  private val maxColumnFamilyId: AtomicInteger = new AtomicInteger(0)
+
+  private val shouldForceSnapshot: AtomicBoolean = new AtomicBoolean(false)
+
+  def getColFamilyNameToIdMap: Map[String, Short] = {
+    colFamilyNameToIdMap.asScala.toMap
+  }
+
+  /**
+   * Check whether the column family name is for internal column families.
+   *
+   * @param cfName - column family name
+   * @return - true if the column family is for internal use, false otherwise
+   */
+  def checkInternalColumnFamilies(cfName: String): Boolean = cfName.charAt(0) == '_'
+
+  /**
+   * Create RocksDB column family, if not created already
+   */
+  def createColFamilyIfAbsent(colFamilyName: String, isInternal: Boolean = false): Short = {
+    if (!checkColFamilyExists(colFamilyName)) {
+      val newColumnFamilyId = maxColumnFamilyId.incrementAndGet().toShort
```

Comment:
Do we not need to set `maxColumnFamilyId` to `newColumnFamilyId` ?

Reply:
Calling `incrementAndGet` sets `maxColFamilyId = maxColFamilyId + 1` and returns that value so I think we're good there